pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -616,24 +608,38 @@ func (p *Pod) constructGroupPodSets() ([]kueue.PodSet, error) {
 		if err != nil {
 			return nil, err
 		}
-		return constructGroupPodSetsFast(p, tc)
+		return constructGroupPodSetsFast(p.list.Items, tc)
 	}
 	return constructGroupPodSets(p.list.Items)
 }
 
-func constructGroupPodSetsFast(p *Pod, groupTotalCount int) ([]kueue.PodSet, error) {
-	for _, podInGroup := range p.list.Items {
+func constructPodSets(p *corev1.Pod) []kueue.PodSet {
+	return []kueue.PodSet{
+		constructPodSet(p),
+	}
+}
+
+func constructPodSet(p *corev1.Pod) kueue.PodSet {
+	return kueue.PodSet{
+		Name:  kueue.DefaultPodSetName,
+		Count: 1,
+		Template: corev1.PodTemplateSpec{
+			Spec: *p.Spec.DeepCopy(),
+		},
+		TopologyRequest: jobframework.PodSetTopologyRequest(&p.ObjectMeta, ptr.To(kueuealpha.PodGroupPodIndexLabel), nil, nil),
```

Comment:
As mentioned , I don't think `PodGroupPodIndexLabel` would work well for rank-based ordering. I suggest to drop it for now, and do properly in a follow up along with e2e tests.

Reply:
This is the pod controller. But you're right, I haven't removed it from the leaderworkerset yet.